Add end-to-end testing scripts and Playwright as a dev dependency

srthkdev opened the pull request:
#2 
- Introduce new scripts for end-to-end testing using Playwright
- Add Playwright as a development dependency in package.json

slavingia commented on tests/e2e/auth.spec.ts:
```diff
@@ -0,0 +1,28 @@
+    // This test verifies the auth page loads properly
+    await expect(page).toHaveTitle(/.*Gumboard.*/);
+  });
+});
```
Always add newlines to files

slavingia commented:
see failing tests

srthkdev commented:
> Doesn't work for me. Does it work for you? See CI

Successfully fixed the test configuration issue by properly excluding Playwright E2E tests from Jest runs.

Changes Made:
- Updated `jest.config.js` with a more specific `testMatch` pattern that only includes Jest unit tests in `lib/__tests__/`
- Added comprehensive `testPathIgnorePatterns` to exclude the entire `tests/` directory where Playwright E2E tests are located

Verification Results:
- ✅ `npm test` now runs only Jest unit tests: 5 tests in `lib/__tests__/utils.test.ts` pass successfully
- ✅ `npm run test:e2e -- --list` shows all 33 Playwright E2E tests are still available and can be run independently
- ✅ No more conflicts between Jest and Playwright test runners

slavingia commented:
Doesn't seem to be added to our CI pipeline so can't tell if it worked or not!

slavingia commented on tests/e2e/auth.spec.ts:
```diff
@@ -0,0 +1,28 @@
+import { test, expect, cleanupTestData } from '../fixtures/test-helpers';
+
+test.describe('Account Creation', () => {
```
You can remove all tests that just test DOM and just test the end-to-end functionality in a single test "user can log in" - same for rest.

slavingia commented on tests/e2e/todo-conversion.spec.ts:
```diff
@@ -0,0 +1,74 @@
+  test('should show demo collaboration features', async ({ page }) => {
+    await page.goto('/');
+    
+    await expect(page.locator('text=Work together seamlessly with your team in real-time')).toBeVisible();
```
For all of these, I would not just arbitrarily test the DOM but specific user interactions and testing the db state is critical to make them e2e tests (vs just front-end tests, which we don't need).

srthkdev commented:
Hi @slavingia  ,

I've been working on addressing both of your feedback points from the PR review and CI issue.

slavingia commented:
It's OK I already merged the other PR
